src/main/java/com/amazon/ion/system/IonReaderBuilder.java

```diff
@@ -340,10 +335,7 @@ private static List<InputStreamInterceptor> detectStreamInterceptorsOnClasspath(
      */
     public List<InputStreamInterceptor> getInputStreamInterceptors() {
```

Comment:
It feels wrong to me that the service-loaded interceptors are not always used.  Those are effectively global-ish configuration, that should have permanent effect.  Attempting to add another interceptor to a local case shouldn't cause the globals to silently disappear.  I would expect surprising defects caused by that side-effect.

If one wants to remove an interceptor locally, or replace/rearrange the global configured list, then a `setInputSTreamInterceptors(List<>)` or somesuch would be reasonable.